### ISSUE TITLE
Trying to run my own API

### ISSUE THREAD

**Issue by jortaca** [open-meteo/open-meteo#278]
First of all, **thank you very much** for this great project. It is exactly what I needed, as I was struggling with gribs trying to get data for my windsurfing app (Vadewind). I found Open-Meteo when I was about to give up, and everything is so easy now. Thanks again!.

I have just started using the hosted API and works absolutely perfect. But I am trying to install it in an old macbook pro, for testing and in case I can make it work fluently, use it, so I don't put weight on yours.

I have a lot of questions, but the main one is that I have a swift error when it starts downloading models. Maybe the Swift version doesn't work with my system? (I'm using MacOs High Sierra 10.13.6). This is what the terminal says:

$ docker-compose run open-meteo download-ecmwf --run 00
Creating open-meteo_open-meteo_run ... done
[ INFO ] Downloading domain ECMWF run '2023-04-03T00:00' (App/DownloadEcmwfCommand.swift:35)
[ INFO ] Downloading hour 0 (App/DownloadEcmwfCommand.swift:64)
[ INFO ] Downloading file https://data.ecmwf.int/forecasts/20230403/00z/0p4-beta/oper/20230403000000-0h-oper-fc.grib2 (App/Curl.swift:81)
Received signal 4. Backtrace:
0x55757a6aa592, Backtrace.(printBacktrace in _B82A8C0ED7C904841114FDF244F9E58E)(signal: Swift.Int32) -> () at /build/.build/checkouts/swift-backtrace/Sources/Backtrace/Backtrace.swift:66
0x55757a6aa827, closure #1 (Swift.Int32) -> () in static Backtrace.Backtrace.install(signals: Swift.Array<Swift.Int32>) -> () at /build/.build/checkouts/swift-backtrace/Sources/Backtrace/Backtrace.swift:80
0x55757a6aa827, @objc closure #1 (Swift.Int32) -> () in static Backtrace.Backtrace.install(signals: Swift.Array<Swift.Int32>) -> () at /build/<compiler-generated>:79
0x7f6a4171951f
0x55757ab36ebd, _p4bits16 at Sources/CTurboPFor/src/vp4c.c:0
0x55757ab40eda, p4nzenc128v16 at Sources/CTurboPFor/src/vp4c.c:418
0x55757af37d13, function signature specialization <Arg[0] = Exploded> of generic specialization <Foundation.FileHandle> of SwiftPFor2D.OmFileWriterState.write(Swift.ArraySlice<Swift.Float>) throws -> () at /build/Sources/SwiftPFor2D/SwiftPFor2D.swift:198
0x55757af3a9ca, generic specialization <Foundation.FileHandle> of SwiftPFor2D.OmFileWriterState.write(Swift.ArraySlice<Swift.Float>) throws -> () at /build/Sources/SwiftPFor2D/SwiftPFor2D.swift:0
0x55757af3a9ca, generic specialization <Foundation.FileHandle> of SwiftPFor2D.OmFileWriter.write<A where A: SwiftPFor2D.OmFileWriterBackend>(fn: A, compressionType: SwiftPFor2D.CompressionType, scalefactor: Swift.Float, supplyChunk: (Swift.Int) throws -> Swift.ArraySlice<Swift.Float>) throws -> () at /build/Sources/SwiftPFor2D/SwiftPFor2D.swift:350
0x55757af3a9ca, function signature specialization <Arg[4] = [Closure Propagated : closure #1 (Swift.Int) -> Swift.ArraySlice<Swift.Float> in SwiftPFor2D.OmFileWriter.write(file: Swift.String, compressionType: SwiftPFor2D.CompressionType, scalefactor: Swift.Float, all: Swift.Array<Swift.Float>) throws -> (), Argument Types : [Swift.Array<Swift.Float>]> of closure #1 () throws -> () in SwiftPFor2D.OmFileWriter.write(file: Swift.String, compressionType: SwiftPFor2D.CompressionType, scalefactor: Swift.Float, supplyChunk: (Swift.Int) throws -> Swift.ArraySlice<Swift.Float>) throws -> () at /build/Sources/SwiftPFor2D/SwiftPFor2D.swift:365
0x55757af3a9ca, function signature specialization <Arg[4] = Owned To Guaranteed> of function signature specialization <Arg[3] = [Closure Propagated : closure #1 (Swift.Int) -> Swift.ArraySlice<Swift.Float> in SwiftPFor2D.OmFileWriter.write(file: Swift.String, compressionType: SwiftPFor2D.CompressionType, scalefactor: Swift.Float, all: Swift.Array<Swift.Float>) throws -> (), Argument Types : [Swift.Array<Swift.Float>]> of SwiftPFor2D.OmFileWriter.write(file: Swift.String, compressionType: SwiftPFor2D.CompressionType, scalefactor: Swift.Float, supplyChunk: (Swift.Int) throws -> Swift.ArraySlice<Swift.Float>) throws -> () at /build/<compiler-generated>:363
0x55757af34da7, function signature specialization <Arg[3] = [Closure Propagated : closure #1 (Swift.Int) -> Swift.ArraySlice<Swift.Float> in SwiftPFor2D.OmFileWriter.write(file: Swift.String, compressionType: SwiftPFor2D.CompressionType, scalefactor: Swift.Float, all: Swift.Array<Swift.Float>) throws -> (), Argument Types : [Swift.Array<Swift.Float>]> of SwiftPFor2D.OmFileWriter.write(file: Swift.String, compressionType: SwiftPFor2D.CompressionType, scalefactor: Swift.Float, supplyChunk: (Swift.Int) throws -> Swift.ArraySlice<Swift.Float>) throws -> () at /build/<compiler-generated>:0
0x55757af34da7, SwiftPFor2D.OmFileWriter.write(file: Swift.String, compressionType: SwiftPFor2D.CompressionType, scalefactor: Swift.Float, all: Swift.Array<Swift.Float>) throws -> () at /build/Sources/SwiftPFor2D/SwiftPFor2D.swift:390
0x55757a329cbd, (3) suspend resume partial function for function signature specialization <Arg[4] = Dead> of App.DownloadEcmwfCommand.downloadEcmwf(application: Vapor.Application, base: Swift.String, run: App.Timestamp, skipFilesIfExisting: Swift.Bool) async throws -> () at /build/Sources/App/Ecmwf/DownloadEcmwfCommand.swift:121
0x7f6a421ea49d
0x7f6a421eac4b
0x7f6a42f3c3f4
0x7f6a42f3c252
0x7f6a42f47a91
0x7f6a4176bb42
0x7f6a417fcbb3
0xffffffffffffffff
ERROR: 132

Apple Swift version 4.2.1 (swiftlang-1000.0.42 clang-1000.10.45.1)
Target: x86_64-apple-darwin17.7.0

Any idea of what it can be? Thanks in advance.

**Comment by patrick-zippenfenig** [comment]
Which CPU generation is inside the MacBook? I develop on M1/M2 and it used to work on my 2018 MacBook. The issue is most likely missing CPU instruction support.

You could try to disable `-march=skylake` in Package.swift

**Comment by jortaca** [comment]
Its quite old... (Mid 2011) Intel Core i7


  Model ID:	MacBookPro8,1
  Processor name:	Intel Core i7
  Processor speed:	2,7 GHz
  Number of processors:	1
  Number of cores:	2
  Caché de nivel 2 (por núcleo):	256 KB
  Caché de nivel 3:	4 MB
  RAM:	8 GB
  Versión de la ROM de arranque:	87.0.0.0.0
  Versión SMC (sistema):	1.68f99

**Comment by patrick-zippenfenig** [comment]
Yeah, the default build target `skylake` is from 2015. It could work if you remove this build target. However, performance might be quite bad...

**Comment by jortaca** [comment]
It didn't work... I think that I'll have to use something more modern :) thanks anyway
May I ask you another question? 
Is it possible to filter the grib downloading by lat and lon location?
The reason why I was trying to use an old machine was that I just need a few variables from 250 locations. So I thought that if I could filter it when downloading the grib, it wouldn't be a lot of data to manage.

**Comment by patrick-zippenfenig** [comment]
Unfortunately, this is not possible. If you do not need data on a global scale, running your own API nodes might be a bit overkill. You can however use the publicly available API. In case you need more then API calls, you can contact me via email
